### DISCUSSION TITLE
Support for highly accurate long reads?

### DISCUSSION THREAD

**Discussion by MrTomRod** [metagenome-atlas/atlas#724] [Ideas]
Dear ATLAS team

We really like your pipeline and have used it for short-read metagenome datasets.

For highly accurate long reads, it seems to us that different assemblers and binners might be necessary.

Are you planning to support such input data in the future? If not, how would you process such data? Maybe use something like [metaMDBG](https://www.nature.com/articles/s41587-023-01983-6), a metagenome-aware assembler plus your suite of quality checks?

Best,
MrTomRod

**Comment by SilasK** [comment]
Hey @MrTomRod, In theory I would like Atlas to support long reads (other than hybrid assembly). But then I don't have the incentive to work on that, as I don't have data myself.

@jmtsuji  made  and extension of Atlas for ONT.  

One option would be simply to use meta-flye as assembler for long-reads only. 
but as you point out there are more and more data types coming in that don't seem to need binning or that can be assembled directly into MAGs.

**Reply by alienzj** [reply]
Here are some publicly available shotgun metagenomics datasets generated with PacBio HiFi sequencing: https://github.com/PacificBiosciences/pb-metagenomics-tools/blob/master/docs/PacBio-Data.md

**Comment by SilasK** [comment]
I will try to implement a long-read only workflow in atlas. 

@LeeBergstrand @jmtsuji  I've seen you made an awesome pipeline for single genome. Can I include/copy some or all of this code in atlas?

**Reply by LeeBergstrand** [reply]
> I will try to implement a long-read only workflow in atlas.
> 
> @LeeBergstrand @jmtsuji I've seen you made an awesome pipeline for single genome. Can I include/copy some or all of this code in atlas?

@SilasK @jmtsuji and I would be interested in this potentially. @jmtsuji Is currently at a conference this week, so we will discuss this and get back to you about this afterwards. My preference would be for you to import this code as part of some sort of library that you would later cite. We are currently modularizing our code into several libraries and tools: https://github.com/orgs/rotary-genomics/repositories. For example, Spokewrench: https://github.com/rotary-genomics/spokewrench

**Reply by jmtsuji** [reply]
@SilasK Sorry for taking so long to reply! I am just getting back to coding tasks now after a busy summer. @LeeBergstrand and I were just talking this week about long read metagenomics and have a few ideas we might like to discuss with you. I'll follow up with you more on this by email. Hope you are well!

**Comment by SilasK** [comment]
Alternatively, I would build something like:


1. **NanoComp / NanoPlot (for an overview of quality):** [NanoComp](https://github.com/wdecoster/nanocomp)
2. **Filtlong (for QC):** [Filtlong](https://github.com/rrwick/Filtlong)
3. **Flye / metaFlye (for assembly):** [Flye](https://github.com/mikolmogorov/Flye)
4. **minimap2 (for mapping):** [minimap2](https://github.com/lh3/minimap2)
5. **racon (read polishing):** [racon](https://github.com/isovic/racon)
6. **medaka (for consensus genome sequence):** [medaka](https://github.com/nanoporetech/medaka)
7. **genometools (gtseqstat for a quick quality check and general statistics):** [genometools](https://github.com/genometools/genometools)

What do you think?

**Comment by MrTomRod** [comment]
> NanoComp / NanoPlot (for an overview of quality): [NanoComp](https://github.com/wdecoster/nanocomp)

Would this pipeline be compatible with PacBio data? Should it be or is ONT the future of metagenomics?

**Reply by LeeBergstrand** [reply]
A lot of the current long read tools will work with both nanopore and pacbio depending on CLI flags because they present similar data.
